packages/polkadart_scale_codec/test/sink_test.dart

```diff
@@ -201,170 +166,427 @@ void main() {
-          args: BigInt.from(0),
-          expectedHex: '0x00000000000000000000000000000000');
+      test(
+          'When u128() is provided with args 0, it must produce result: 0x00000000000000000000000000000000.',
+          () {
+        final expectedResult = '0x00000000000000000000000000000000';
+
+        final encoder = HexEncoder();
+        encoder.u128(BigInt.from(0));
+
+        final computedHex = encoder.toHex();
+
+        expect(computedHex, expectedResult);
+      });
 
       // highest acceptable possible
-      testPrimitiveCompact(
-          methodName: 'u128',
-          args: 2.toBigInt.pow(128) - 1.toBigInt,
-          expectedHex: '0xffffffffffffffffffffffffffffffff');
+      test(
+          'When u128() is provided with args 340282366920938463463374607431768211455, it must produce result: 0xffffffffffffffffffffffffffffffff.',
+          () {
+        final expectedResult = '0xffffffffffffffffffffffffffffffff';
+
+        final encoder = HexEncoder();
+        encoder.u128(2.toBigInt.pow(128) - 1.toBigInt);
+
+        final computedHex = encoder.toHex();
+
+        expect(computedHex, expectedResult);
+      });
     }
 
     // u256
     {
       // lowest acceptable possible
-      testPrimitiveCompact(
-          methodName: 'u256',
-          args: BigInt.from(0),
-          expectedHex:
-              '0x0000000000000000000000000000000000000000000000000000000000000000');
+      test(
+          'When u256() is provided with args 0, it must produce result: 0x0000000000000000000000000000000000000000000000000000000000000000.',
+          () {
+        final expectedResult =
+            '0x0000000000000000000000000000000000000000000000000000000000000000';
+
+        final encoder = HexEncoder();
+        encoder.u256(BigInt.from(0));
+
+        final computedHex = encoder.toHex();
+
+        expect(computedHex, expectedResult);
+      });
 
       // highest acceptable possible
-      testPrimitiveCompact(
-          methodName: 'u256',
-          args: 2.toBigInt.pow(128) - 1.toBigInt,
-          expectedHex:
-              '0xffffffffffffffffffffffffffffffff00000000000000000000000000000000');
+      test(
+          'When u256() is provided with args 115792089237316195423570985008687907853269984665640564039457584007913129639935, it must produce result: 0xffffffffffffffffffffffffffffffff00000000000000000000000000000000.',
+          () {
+        final expectedResult =
+            '0xffffffffffffffffffffffffffffffff00000000000000000000000000000000';
+
+        final encoder = HexEncoder();
+        encoder.u256(2.toBigInt.pow(128) - 1.toBigInt);
+
+        final computedHex = encoder.toHex();
+
+        expect(computedHex, expectedResult);
+      });
     }
 
     // Signed Bits
     //i8
     {
       // i8 lowest acceptable possible
-      testPrimitiveCompact(methodName: 'i8', args: -128, expectedHex: '0x80');
+      test(
+          'When i8() is provided with args -128, it must produce result: 0x80.',
```

Comment:
nit: have the comments explain your tests, and you don't need comments + your code gets self documented. This applies to all tests
```suggestion
          'it should convert lowest acceptable possible 8 bit value',
```